shard.lua

```diff
@@ -775,23 +783,22 @@ local function append_shard(servers, is_replica, start_waiter)
                 id = pool.zones_n, n = 0, list = {}
             }
         end
-        -- disable new shard by default
-        --server.ignore = true
+
         -- append server to connection pool
         pool:connect((shards_n + 1) * redundancy - id + 1, server)
         local zone = pool.servers[server.zone]
         zone.list[zone.n].zone_name = server.zone
 
         if not server.arbiter then
+            shards[shards_n + 1] = shards[shards_n + 1] or {}
```

Comment:
why did you move it here? if we server is arbiter, then `shards_n` will increment in line 796 without creating a shard. On the other hand, I wonder what arbiter is, and whether shard can become one...